### ISSUE TITLE
Make compatible with multiprocess or rewrite to WebExtensions

### ISSUE THREAD

**Issue by GoodGuyNick** [antoniy/mozilla-youtube-dl-binding#40]
Extension incompatible with Firefox multiprocess. You can check it using [Add-on Compatibility Reporter](https://addons.mozilla.org/en-US/firefox/addon/add-on-compatibility-reporter/).
[Update on Compatibility Milestones](https://blog.mozilla.org/addons/2017/03/27/update-compatibility-milestones/):

> Firefox won’t run in multiprocess mode unless all enabled add-ons have the [multiprocessCompatible flag](https://developer.mozilla.org/en-US/Add-ons/Install_Manifests#multiprocessCompatible) set to true or are WebExtensions. This means that developers who haven’t set this flag don’t have to worry about multiprocess compatibility and can focus on WebExtensions and the Firefox 57 deadline.

**Comment by utack** [comment]
Seems to be working when you force it  
https://wiki.mozilla.org/Electrolysis#Force_Enable  
But I agree it would be nice to have it working for everyone who does not know about this or does not want to use this hack

**Comment by Keith94** [comment]
It'd be great to see plans on migrating to WebExtensions.

The add-on requires toggling `extensions.allow-non-mpc-extensions` to `true` to install in Fx 57.

**Comment by skorokithakis** [comment]
This extension makes my browsing ten times better, please don't let it die with FF 56, @antoniy.

**Comment by monochrony** [comment]
alternatively, you can drag urls out of links or the adressbar directly into MPV to make it work. holding shift while doing so even adds files to the playlist.

isn't as instantaneous, as you have to open the player first, but better than nothing.

**Comment by dummys** [comment]
I have a poc running right now, need to clean it a bit and I will send a pull request.
Basically I use a python wrapper as a native app to call the selected player with subprocess.run().

**Comment by GoodGuyNick** [comment]
I think this project is dead so I've switched to [Open With](https://addons.mozilla.org/en-US/firefox/addon/open-with/).

**Comment by maxime-michel** [comment]
Thanks for the tip @GoodGuyNick, there's no keyboard shortcut with that option though.

**Comment by dummys** [comment]
I pull requested one @ open_with github, checkit

**Comment by maxime-michel** [comment]
Yeah, I did see that. However, I'm switching to qutebrowser. It might not float everybody's boat, but as a Vimmer I like it. And that plugin is a one-liner with it:

    config.bind(',m', 'spawn /usr/local/bin/mpv {url} --pause --keep-open')

**Comment by woodruffw** [comment]
A bit late to the party, but I created [ff2mpv](https://github.com/woodruffw/ff2mpv) as a WebExtension clone for mpv-youtube-dl-binding. It provides the same context menu/toolbar interface as this plugin did, and nothing else.
